extensions/systemd/libwrapper/DlopenWrapper.cpp

```diff
@@ -72,7 +72,7 @@ class DlopenJournal : public Journal {
   template<typename F>
   F loadSymbol(const char* const symbol_name) {
     // The cast below is supported by POSIX platforms. https://stackoverflow.com/a/1096349
-    F const symbol = (F)dlsym(libhandle_.get(), symbol_name);
+    F const symbol = reinterpret_cast<F>(dlsym(libhandle_.get(), symbol_name));
```

Comment:
A cast from object to function pointers is normally undefined behavior, but in the case of dlsym, this is allowed on POSIX. I'm not sure that a `reinterpret_cast` is guaranteed to do the right thing in this case. I'd rather keep the C-style cast here, because that's guaranteed to work, while with `reinterpret_cast`, an implementation may decide later that it's undefined behavior to do this cast even with `dlsym`.

Reply:
Good point, updated in 9d15d87a77f8e03e0863f8910fc540d9665b9d8f